src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html

```diff
@@ -1,260 +1,206 @@
-                       html="{{ passwordPolicyHelpText }}">
-            </cd-helper>
-          </label>
-          <div class="cd-col-form-input">
-            <div class="input-group">
-              <input class="form-control"
-                     type="password"
-                     placeholder="Password..."
-                     id="password"
-                     name="password"
-                     autocomplete="new-password"
-                     formControlName="password">
-              <button type="button"
-                      class="btn btn-light"
-                      cdPasswordButton="password">
-              </button>
-            </div>
-            <div class="password-strength-level">
-              <div class="{{ passwordStrengthLevelClass }}"
-                   data-toggle="tooltip"
-                   title="{{ passwordValuation }}">
-              </div>
-            </div>
-            <span class="invalid-feedback"
-                  *ngIf="userForm.showError('password', formDir, 'required')"
-                  i18n>This field is required.</span>
-            <span class="invalid-feedback"
-                  *ngIf="userForm.showError('password', formDir, 'passwordPolicy')">
-              {{ passwordValuation }}
+      <!-- Confirm password -->
+      <div class="form-item">
+        <cds-password-label for="confirmpassword"
+                            label="Confirm password..."
+                            [invalid]="!userForm.controls.confirmpassword.valid && userForm.controls.confirmpassword.dirty"
+                            [invalidText]="confirmpasswordError"
+                            i18n>
+          <input cdsPassword
+                 type="password"
+                 placeholder="Confirm password..."
+                 id="confirmpassword"
+                 name="confirmpassword"
+                 formControlName="confirmpassword">
+        </cds-password-label>
+        <ng-template #confirmpasswordError>
+          <span class="invalid-feedback"
+                *ngIf="userForm.showError('confirmpassword', formDir, 'match')"
+                i18n>Password confirmation doesn't match the password.</span>
+          <span class="invalid-feedback"
+                *ngIf="userForm.showError('confirmpassword', formDir, 'required')"
+                i18n>This field is required.</span>
+        </ng-template>
+      </div>
+      <!-- Password expiration date -->
+      <div class="form-item"
+           *ngIf="!authStorageService.isSSO()">
+        <cd-date-time-picker [ngClass]="{'required': pwdExpirationSettings.pwdExpirationSpan > 0}"
+                             [control]="userForm.get('pwdExpirationDate')"
+                             name="Password expiration date"
+                             [hasTime]="false"
+                             [helperText]="helperText"
+                             i18n-name
+                             >
+        </cd-date-time-picker>
```

Comment:
by default, we are selecting today's date which means that whatever new users that are created are getting created with an expiration of today which is not intended. This field is optional so this should be set to empty at the beginning and once the user sets a value then only this should be updated. I know the date-time-picker is populating the value onInit. can we adapt that component, by introducing a toggle, in a way that its empty at first and it get's the value later on?

![Screenshot from 2024-09-18 10-00-55](https://github.com/user-attachments/assets/2176f5cc-88bc-4c87-bb1d-fddf74dd4475)

Reply:
![image](https://github.com/user-attachments/assets/0b73a509-7311-4010-be27-d9b87cd7c8af)